myhpi/core/templates/core/toc_button.html

```diff
@@ -4,12 +4,12 @@
 {% load i18n %}
 {% load bootstrap_icons %}
 
-{%if toc|hasTocContent %}
+{%if toc|hasTocContent or attachments %}
     <div id="sidebar-toggle" class="d-grid d-lg-none d-print-none">
         <button type="button" data-bs-toggle="offcanvas" data-bs-target="#sidebar-offcanvas"
-            aria-controls="sidebar-offcanvas" class="btn btn-light mb-3" aria-label="{% translate 'Table of contents' %}">
```

Comment:
you could also customize the text to only show what is actually there (just "ToC" if there are not attachements)